### PR TITLE
ci(codeql): ignore scanning of generated files

### PR DESCRIPTION
**Pull request by Camc314**
Ignores CodeQl analysis of *.md and files in api/*

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -3,8 +3,14 @@ name: 'CodeQL'
...
   push:
     branches: [master]
+    paths-ignore:
+      - api/*
   pull_request:
     branches: [master]
+    paths-ignore:
+      - '**/*.md'
+    paths-ignore:
+      - 'api/**'
```
Comment (h1dden-da3m0n) [inline .github/workflows/codeql-analysis.yml:13]: u have the `paths-ignore` key there twice?

Comment (h1dden-da3m0n) [reply .github/workflows/codeql-analysis.yml:13]:
```suggestion
    paths-ignore:
      - '**/*.md'
      - 'api/**'
```
I think you wanted to do this instead?

Comment (Camc314) [reply .github/workflows/codeql-analysis.yml]: Ah yeah. Good catch, my mistake 🤦‍♀️

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -3,8 +3,13 @@ name: 'CodeQL'
 on:
   push:
     branches: [master]
+    paths-ignore:
+      - api/*
```
Comment (MrTimscampi) [inline .github/workflows/codeql-analysis.yml:7]: Shouldn't this be identical to the ones on pull requests?

Comment (Camc314) [reply .github/workflows/codeql-analysis.yml]: Yep. I don't know why I did them differently.

**Review by h1dden-da3m0n — APPROVED** [review]

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**